From 5ad0f0aa4979e98dfa3ff02dc10bd14c05ecbed9 Mon Sep 17 00:00:00 2001
From: xiaohuin <xiaohuin@codeaurora.org>
Date: Thu, 1 Mar 2018 16:01:37 +0800
Subject: [PATCH] Accessibility: Fix FWR when unlinkToDeath if linkToDeath failed

If linkToDeath failed, NoSuchElementException will throw when
unlinkToDeath. Catch this exception to avoid system server reboot

diff --git a/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java b/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java
index 83dfccb57eb..76f1515e197 100644
--- a/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java
+++ b/services/accessibility/java/com/android/server/accessibility/AccessibilityManagerService.java
@@ -127,6 +127,7 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.NoSuchElementException;
 import java.util.Set;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.function.Consumer;
@@ -3687,7 +3688,11 @@ public void resetLocked() {
                 /* ignore */
             }
             if (mService != null) {
-                mService.unlinkToDeath(this, 0);
+                try {
+                    mService.unlinkToDeath(this, 0);
+                } catch (NoSuchElementException e) {
+                    Slog.e(LOG_TAG, "Unable to unlinkToDeath", e);
+                }
                 mService = null;
             }
             mServiceInterface = null;
